solutions/cf-1557B.mdx

```diff
@@ -0,0 +1,347 @@
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>value</th>
+			<th>1</th>
+			<th>2</th>
+			<th>3</th>
+			<th>4</th>
+			<th>6</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{map}[\texttt{value}]$</td>
+			<td>$\infty$</td>
+			<td>1</td>
+			<td>2</td>
+			<td>3</td>
+			<td>4</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+With this map in mind, we generate the subsets as shown:
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th>1</th>
+			<th>2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th>2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th className="bg-red-100 dark:bg-red-800">3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th className="bg-red-100 dark:bg-red-800">3</th>
+			<th className="bg-blue-100 dark:bg-blue-800">4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+Thus, we need 4 subarrays at minimum.
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<Spoiler title="Time Complexity Proof">
+Sorting Requires $\mathcal{O}(N\log N)$, and repeated map insertions also require $\mathcal{O}(N\log N)$. Hence the overall time complexity is $\mathcal{O}(N\log N)$.
+</Spoiler>
```

Comment:
is this spoiler necessary?

Reply:
It makes it look better. But I can't even see the output because it is not coming up in the search bar. If you check the deployment and search for it in the problems tab, it doesn't come up and I don't know why